lightkurve/mast.py

```diff
@@ -61,86 +69,242 @@ def search_kepler_products(target):
             target_name = 'ktwo{:09d}'.format(target)
         else:
             raise ValueError("Not in the Kepler KIC or EPIC ID range")
-        obs = Observations.query_criteria(target_name=target_name,
+        obs = Observations.query_criteria(objectname=target_name,
+                                          radius='{} arcsec'.format(radius),
                                           project=["Kepler", "K2"])
+        #Remove KeplerFFI data set.
+        obs = obs[(obs['obs_collection']=='Kepler') | (obs['obs_collection']=='K2')]
+        obs.sort('distance')
     except ValueError:
         # If `target` did not look like a KIC or EPIC ID, then we let MAST
         # resolve the target name to a sky position.
         try:
             obs = Observations.query_criteria(objectname=target,
-                                              radius='1 arcsec',
+                                              radius='{} arcsec'.format(radius),
                                               project=["Kepler", "K2"])
+            #Remove KeplerFFI data set.
+            obs = obs[(obs['obs_collection']=='Kepler') | (obs['obs_collection']=='K2')]
+            obs.sort('distance')
         except ResolverError as exc:
             raise ArchiveError(exc)
-    return Observations.get_product_list(obs)
+
+    #Make sure the final table is in DISTANCE order.
+    obsids = np.asarray(obs['obsid'])
+    t = (Observations.get_product_list(obs))
+    l = [np.where(t['parent_obsid'] == o)[0]  for o in obsids]
+    l = [item for sublist in l for item in sublist]
+    t = t[l]
```

Comment:
Please add a comment explaining the purpose of the above block (I don't understand it).  Also replace `t` and `l` by more verbose variable names to help clarify.